### PR TITLE
Refactoring documenation for commands/flags

### LINKED ISSUE #1360: Clarify preact build --help message
**Do you want to request a _feature_ or report a _bug_?**
Bug

**What is the current behaviour?**
Tool prints the possible commands, and defaults when I type `preact build --help`, but it doens't tell how to disable `default true` options. It should write (as on github) that the way to do so is by attaching `--no-`. It is not obvious and attempts such as ` preact build --sw false --esm false ` give the false error:
```
✖ ERROR /home/deniz/.nvm/versions/node/v12.13.0/lib/node_modules/preact-cli/lib/lib/entry.js
Module not found: Error: Can't resolve 'preact-cli-entrypoint' in '/home/deniz/.nvm/versions/node/v12.13.0/lib/node_modules/preact-cli/lib/lib'
 @ /home/deniz/.nvm/versions/node/v12.13.0/lib/node_modules/preact-cli/lib/lib/entry.js 32:25-57 38:29-61
```

**If the current behaviour is a bug, please provide the steps to reproduce.**
`preact build --no-prerender true --sw false --esm false --inline-css false`

**What is the expected behaviour?**
See above

**If this is a feature request, what is motivation or use case for changing the behaviour?**

**Please mention other relevant information.**

Please paste the results of `preact info` here.
```
System:
    OS: Linux 5.7 Manjaro Linux
    CPU: (4) x64 Intel(R) Core(TM) i3-2370M CPU @ 2.40GHz
  Binaries:
    Node: 12.13.0 - ~/.nvm/versions/node/v12.13.0/bin/node
    npm: 6.12.0 - ~/.nvm/versions/node/v12.13.0/bin/npm
  Browsers:
    Firefox: 78.0.2
  npmPackages:
    preact: ^10.1.0 => 10.4.6 
    preact-cli: ^3.0.0 => 3.0.1 
    preact-render-to-string: ^5.1.2 => 5.1.10 
  npmGlobalPackages:
    preact-cli: 3.0.1
```

### PR DESCRIPTION
**Pull request by RyanChristian4427**
**What kind of change does this PR introduce?**

Mostly simple refactoring

**Did you add tests for your changes?**

N/A

**Summary**

As I went to fix #1360, as it is a valid concern, I noticed the documentation was a bit messy in areas. The flags for `build` vs `watch` were in completely different orders, 1-2 flags had incorrect defaults listed in the README, some valid flags weren't documented at all, and there were grammatical inconsistencies throughout. This PR fixes those issues. 

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,16 +118,17 @@ $ preact build
...
     --template         Path to custom HTML template
     --preload          Adds preload tags to the document its assets  (default false)
     --analyze          Launch interactive Analyzer to inspect production bundle(s)
     --prerenderUrls    Path to pre-rendered routes config  (default prerender-urls.json)
+    --brotli           Adds brotli redirects to the service worker  (default false)
+    --inline-css       Adds critical css to the prerendered markup  (default true)
     -c, --config       Path to custom CLI config  (default preact.config.js)
-    --esm              Builds ES-2015 bundles for your code.  (default true)
-    --brotli           Adds brotli redirects to the service worker.  (default false)
-    --inline-css       Adds critical css to the prerendered markup.  (default true)
```
Comment (RyanChristian4427) [inline README.md:130]: I moved these flags to match the order of `watch` as I felt its was better. Having consistent ordering between the commands improves readability.

File: README.md [hunk 2]
```diff
@@ -118,16 +118,17 @@ $ preact build
     --src              Specify source directory  (default src)
     --dest             Specify output directory  (default build)
     --cwd              A directory to use instead of $PWD  (default .)
+    --esm              Builds ES-2015 bundles for your code  (default true)
     --sw               Generate and attach a Service Worker  (default true)
+    --babelConfig      Path to custom Babel config (default .babelrc)
```
Comment (RyanChristian4427) [inline README.md:123]: This flag seems to have been added back in #1052 but no documentation was added to the ReadMe for it.

File: README.md [hunk 3]
```diff
@@ -139,22 +140,24 @@ Spin up a development server with multiple features like `hot-module-replacement
...
+    --esm              Builds ES-2015 bundles for your code  (default false)
+    --sw               Generate and attach a Service Worker  (default false)
+    --babelConfig      Path to custom Babel config (default .babelrc)
+    --json             Generate build stats for bundle analysis
+    --https            Run server with HTTPS protocol
+    --key              Path to PEM key for custom SSL certificate
+    --cert             Path to custom SSL certificate
+    --cacert           Path to optional CA certificate override
+    --prerender        Pre-render static content on first run
+    --prerenderUrls    Path to pre-rendered routes config  (default prerender-urls.json)
```
Comment (RyanChristian4427) [inline README.md:154]: Like with the `--babelConfig` flag, [this seems to be a supported option](https://github.com/preactjs/preact-cli/blob/e486349e36f27ea52feefb28536d601748a3699e/packages/cli/lib/index.js#L100) that just never got added to the ReadMe

File: packages/cli/lib/index.js [hunk 4]
```diff
@@ -32,19 +32,18 @@ let prog = sade('preact').version(pkg.version);
 
 prog
 	.command('build [src]')
-	.describe('Create a production build')
+	.describe(
+		'Create a production build. You can disable "default: true" flags by prefixing them with --no-<option>'
+	)
```
Comment (RyanChristian4427) [inline packages/cli/lib/index.js:37]: This should close #1360.

File: packages/cli/lib/index.js [hunk 5]
```diff
@@ -32,19 +32,18 @@ let prog = sade('preact').version(pkg.version);
...
 	.option('--sw', 'Generate and attach a Service Worker', true)
+	.option('--babelConfig', 'Path to custom Babel config', '.babelrc')
 	.option('--json', 'Generate build stats for bundle analysis')
 	.option('--template', 'Path to custom HTML template')
 	.option('--preload', 'Adds preload tags to the document its assets', false)
-	.option(
-		'--refresh',
-		'Enables experimental preact-refresh functionality',
-		false
-	)
```
Comment (RyanChristian4427) [inline packages/cli/lib/index.js:47]: Unless I'm mistaken, this was a mistake? I believe this should go under the `watch` command so I moved it there.

File: packages/cli/lib/index.js [hunk 6]
```diff
@@ -82,26 +79,31 @@ prog
 	.describe('Start a live-reload server for development')
 	.option('--src', 'Specify source directory', 'src')
 	.option('--cwd', 'A directory to use instead of $PWD', '.')
-	.option('--esm', 'Builds ES-2015 bundles for your code.', false)
+	.option('--esm', 'Builds ES-2015 bundles for your code', false)
 	.option('--clear', 'Clear the console', true)
 	.option('--sw', 'Generate and attach a Service Worker', undefined)
-	.option('--babelConfig', 'Specify the babel config file', '.babelrc')
+	.option('--babelConfig', 'Path to custom Babel config', '.babelrc')
```
Comment (RyanChristian4427) [inline packages/cli/lib/index.js:85]: Changed the verbiage here purely for consistency. 

"Path to PEM key for custom SSL..."
"Path to custom SSL..."
"Path to custom HTML..."
"Path to custom CLI..."

"Path to custom Babel config" seems to fit better.

File: README.md [hunk 7]
```diff
@@ -139,22 +140,24 @@ Spin up a development server with multiple features like `hot-module-replacement
...
-    --cert          Path to custom SSL certificate
-    --cacert        Path to optional CA certificate override
-    --prerender     Pre-render static content on first run
-    --template      Path to custom HTML template
-    --refresh       Will use [`Preact-refresh`](https://github.com/JoviDeCroock/preact-refresh) to do hot-reloading
-    -c, --config    Path to custom CLI config  (default preact.config.js)
-    -H, --host      Set server hostname  (default 0.0.0.0)
-    -p, --port      Set server port  (default 8080)
-    -h, --help      Displays this message
+    --src              Specify source directory  (default src)
```
Comment (RyanChristian4427) [inline README.md:143]: I added more spacing before the flag descriptions, firstly, to match the spacing found under `build`, and secondly the `prerenderUrls` flag was scraping it once added. Extra spacing helped with the readability.

**Review by ForsakenHarmony — APPROVED** [review]

**Comment by ForsakenHarmony** [comment]
Thanks for the PR @RyanChristian4427 

Seems it got missed, sorry about that

**Comment by RyanChristian4427** [comment]
Oh, certainly nothing pressing about this. 

Thanks for the merge!

**Pull request merged**